### PR TITLE
Fixes #115 - Verify downloads using SHA-256 hashes

### LINKED ISSUE #115: Verify software downloads using SHA-256 hashes
Uno currently uses MD5 hashes to verify software.  However, Muchos uses SHA-256 due to a requirement by Ansible.  It would be nice if both Uno & Muchos used the same hash algorithm.

### PR DESCRIPTION
**Pull request by mikewalch**

### CODE REVIEW

File: bin/impl/load-env.sh [hunk 1]
```diff
@@ -120,14 +120,13 @@ fi
 : "${YARN_LOG_DIR:?"YARN_LOG_DIR is not set in env.sh"}"
 : "${ZOO_LOG_DIR:?"ZOO_LOG_DIR is not set in env.sh"}"
 
+hash shasum 2>/dev/null || { echo >&2 "shasum must be installed & on PATH.  Aborting."; exit 1; }
+export HASH_CMD="shasum -a 256"
```
Comment (keith-turner) [inline bin/impl/load-env.sh]: so this command is the same on Mac?

File: bin/impl/util.sh [hunk 2]
```diff
@@ -16,15 +16,15 @@
...
+  expected_hash=$2
+  actual_hash=$($HASH_CMD "$DOWNLOADS/$tarball" | awk '{print $1}')
 
   if [[ ! -f "$DOWNLOADS/$tarball" ]]; then
     echo "The tarball $tarball does not exists in downloads/"
     exit 1
   fi
-  if [[ "$actual_md5" != "$expected_md5" ]]; then
-    echo "The MD5 checksum ($actual_md5) of $tarball does not match the expected checksum ($expected_md5)"
+  if [[ "$actual_hash" != "$expected_hash" ]]; then
```
Comment (keith-turner) [inline bin/impl/util.sh]: there is another place in the code where hashes are compared and it also checks length... this is a nice sanity check to defend against odd cases like comparing  two empty strings.....

File: bin/impl/fetch.sh [hunk 3]
```diff
@@ -17,36 +17,36 @@
...
   url_prefix=$1
   tarball=$2
-  expected_md5=$3
+  expected_hash=$3
 
-  exp_md5_len=${#expected_md5}
-  if [[ "$exp_md5_len" -ne 32 ]]; then
-    echo "The expected MD5 checksum ($expected_md5) of $tarball has a length of $exp_md5_len but should be 32"
+  exp_hash_len=${#expected_hash}
+  if [[ "$exp_hash_len" -ne $HASH_LEN ]]; then
```
Comment (keith-turner) [inline bin/impl/fetch.sh]: could have a stricter sanity check here using a regex, verify its hex and of expected len.....

**Comment by keith-turner** [comment]
+1

File: bin/impl/load-env.sh (hunk 1 above)
Comment (mikewalch) [inline bin/impl/load-env.sh]: yes it is

File: bin/impl/util.sh (hunk 2 above)
Comment (mikewalch) [inline bin/impl/util.sh]: added in 738ccdfd3

File: bin/impl/fetch.sh (hunk 3 above)
Comment (mikewalch) [inline bin/impl/fetch.sh]: added regex check in 738ccdfd3

**Pull request merged**